cfme/v2v/migrations.py

```diff
@@ -360,21 +381,68 @@ def plan_in_progress(self, plan_name):
             return False
 
 
+class MigrationDashboardView59z(MigrationDashboardView):
+    """Dashboard for 59z has infra_mapping_list while 510z moves it to a separate page.
+    Hence, Inheritance."""
+    infra_mapping_list = InfraMappingList("infra-mappings-list-view")
+    migr_dropdown = MigrationDropdown(text="Not Started Plans")
+
+    @property
+    def is_displayed(self):
+        return (self.navigation.currently_selected == ['Compute', 'Migration'] and
+            (len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0) and
+            (len(self.browser.elements('.//div[contains(@class,"card-pf")]')) > 0))
+
+    def switch_to(self, section):
+        """Switches to Not Started, In Progress, Complete or Archived Plans section."""
+        self.logger.info("Switching to Migration Dashboard section: %s", section)
+        self.migr_dropdown.item_select(section)
+
+
+class InfrastructureMappingView(BaseLoggedInPage):
+    """This is an entire separate page, with many elements similar to what we had in
+    MigrationPlanRequestDetailsView , so re-using some of those Paginator things
+    by renaming those from v2vPaginator to v2vPaginator, etc."""
+
+    infra_mapping_list = InfraMappingList("infra-mappings-list-view")
+    create_infrastructure_mapping = Text(locator='(//a|//button)'
+                                                 '[text()="Create Infrastructure Mapping"]')
+    configure_providers = Text(locator='//a[text()="Configure Providers"]')
+    paginator_view = View.include(v2vPaginatorPane, use_parent=True)
+    search_box = TextInput(locator=".//div[contains(@class,'input-group')]/input")
+    clear_filters = Text(".//a[text()='Clear All Filters']")
+    # Used for Ascending/Descending sort
+    sort_order = Text(".//button[./span[contains(@class,'sort-direction')]]")
+    # Used to select filter_by 'Name' or 'Description'
+    filter_by_dropdown = SelectorDropdown('id', 'filterFieldTypeMenu')
+    # USed to select sort by options like 'Name', 'Number of Associated Plans'
+    sort_by_dropdown = SelectorDropdown('id', 'sortTypeMenu')
+
+    @property
+    def is_displayed(self):
+        # TODO: Resmove 1st condition, once /manageiq-v2v/issues/768 fix is backported to 510z
+        return ((self.navigation.currently_selected ==
+            ['Compute', 'Migration', 'Overview'] or self.navigation.currently_selected ==
+            ['Compute', 'Migration', 'Infrastructure Mappings']) and
+            len(self.browser.elements(".//div[contains(@class,'spinner')]")) == 0 and
+            (self.create_infrastructure_mapping.is_displayed or
+                self.infra_mapping_list.is_displayed or self.configure_providers.is_displayed))
+
+
 class AddInfrastructureMappingView(View):
     form = InfraMappingWizard()
+    form_title_text = VersionPicker({
+        Version.lowest(): 'Infrastructure Mapping Wizard',
+        '5.10': 'Create Infrastructure Mapping'
+    })
```

Comment:
**Question** Why did you move this out, was this going to be used in another method?

Reply:
This was failing otherwise. Mike did ask me to debug why, but I didn't get a chance, but current way seems to work fine.